### PR TITLE
02.07

### PR DESCRIPTION
**Pull request by Pozharitskiy**

### CODE REVIEW

File: 02.07/4.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+curry = function (fn) {
```
Comment (dlos-hiqo) [inline 02.07/4.js:1]: Please rewrite using .bind

File: 02.07/5.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+function debounce(fn, timeOut) {
+    if (timerId) {
+        clearTimeout(timerId);
+    }
+    timerId = setTimeout(fn, timeOut)
+}
+let timerId;
```
Comment (dlos-hiqo) [inline 02.07/5.js]: Please rewrite using closure

File: 02.07/6.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    let cache = {};
+    return (...args) => {
+        let n = args;
+        if (n in cache) {
+            console.log('I have done it by noon', fn(...args));
+            return fn(...args);
+        } else {
+            console.log('Calculating', fn(...args));
+            let result = (fn(...args));
+            cache[n] = result;
```
Comment (dlos-hiqo) [inline 02.07/6.js]: It will not work as expected. Please use hash for parameters

**Review by dlos-hiqo — CHANGES_REQUESTED** [review]

File: 02.07/4.js (hunk 1 above)
Comment (Pozharitskiy) [reply 02.07/4.js:1]: Done

File: 02.07/5.js (hunk 2 above)
Comment (Pozharitskiy) [reply 02.07/5.js]: Done

File: 02.07/6.js (hunk 3 above)
Comment (Pozharitskiy) [reply 02.07/6.js]: Done

**Pull request merged**